### ISSUE TITLE
Question about loss_cls.

### ISSUE THREAD

**Issue by Tu1016** [open-mmlab/mmdetection3d#500]
Why does the cls_target include three values([-1, 0, 1]) when I only train a car detector on kitti dataset? How does FocalLoss work?

**Comment by Tai-Wang** [comment]
Typically -1 means `Dont Care`, 0 means the only one class and 1 means the background.

**Comment by Tu1016** [comment]
@Tai-Wang Hi, thank you firstly! And will the anchors corresponding to -1 contribute to cls_loss when SigmoidFocalLoss consumes the cls_pred and this kind of cls_target? If not, how do you make that? Actually, I can not understand how you calculate the loss.

**Comment by Tai-Wang** [comment]
Yes. When computing the loss for classification, the negative samples will also be considered. Afterwards, when computing losses for regression or others, we usually only consider positive samples.

**Comment by Tu1016** [comment]
@Tai-Wang You mean finally the net is supposed to output labels range from -1 to 1?

**Comment by Tai-Wang** [comment]
The classification output is a normalized vector after softmax, with which you can determine whether a specific predicition is a foreground one or background. It is basically the same as 2D detection.

**Comment by Tu1016** [comment]
Sorry, I am still confused. I want to illustrate my problem in details. When I train a car detector, the number of cls_pred channel is one. After sigmoid function, cls_pred values’ range is 0~1, so I think the cls_target should only includes 0 and 1. So the -1 in cls_target makes me very confused, the net can not output -1. Looking for your answer, thank you very much.

**Comment by Wuziyi616** [comment]
No. -1 is just some flag for DONT CARE objects and it won't be used in cls_loss such as CrossEntropy or BCELoss, which means it will be treated similar to those with label 0. I haven't dug deeper into the code for box gt generation, but I guess we only take those `label==1` as positive and all other labels (-1 and 0) will be assigned as negative.

**Comment by Tu1016** [comment]
@Wuziyi616 Thanks a lot. Now I know the net treats -1 as negative.

**Comment by Wuziyi616** [comment]
Glad it helps! Feel free to reopen this issue if you have any further questions :)
